schema-engine/sql-schema-describer/src/mysql/constraints_query.sql

```diff
@@ -0,0 +1,12 @@
+SELECT
+	tc.table_schema AS namespace,
+	tc.table_name AS table_name,
+	tc.constraint_name AS constraint_name,
+	CAST(LOWER(LEFT(tc.constraint_type, 1)) AS CHAR) AS constraint_type,
```

Comment:
This is a tiny optimization to only compare one character, rather than a string of several characters. It's non-ambiguous, and is inspired by how Postgres distinguishes the constraint types (e.g., "c" for "CHECK", "x" for "EXCLUDE" constraints).

Reply:
I'd prefer to keep the whole string there, because MySQL doesn't guarantee that they won't add a new constraint type that starts with a "C" in the future, whereas Postgres does.

Reply:
Same. If MySQL has full strings that mean something, let's use those instead of trying to be smart to safe a few nanoseconds.